### PR TITLE
Labeling algorithm updates for C API

### PR DESCRIPTION
**Pull request by ChuckHastings**
Several things bundled here:
 * Added C API definition, implementation and tests for weakly connected components
 * Added C API definition for strongly connected components
 * Updated the C API test functions for supporting the `is_symmetric` property
 * Added C unit tests for SCC, and an implementation that returns `NOT_IMPLEMENTED`

I could add some mock data returns, although the output from SCC is the same as the output from WCC.  So python testing could just call WCC instead until there is an SCC implementation.

### CODE REVIEW

File: cpp/src/c_api/strongly_connected_components.cpp [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            typename weight_t,
+            bool store_transposed,
+            bool multi_gpu>
+  void operator()()
+  {
+    // FIXME: Think about how to handle SG vice MG
+    if constexpr (!cugraph::is_candidate<vertex_t, edge_t, weight_t>::value) {
+      unsupported();
+    } else {
+      // WCC expects store_transposed == false
```
Comment (seunghwak) [inline cpp/src/c_api/strongly_connected_components.cpp:61]: WCC=>SCC

File: cpp/src/c_api/weakly_connected_components.cpp [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+#include <cugraph/graph_functions.hpp>
+
+#include <optional>
+
+namespace {
+
+struct wcc_functor : public cugraph::c_api::abstract_functor {
+  raft::handle_t const& handle_;
+  cugraph::c_api::cugraph_graph_t* graph_;
+  bool do_expensive_check_;
```
Comment (seunghwak) [inline cpp/src/c_api/weakly_connected_components.cpp:37]: Add "{}"

File: cpp/src/c_api/strongly_connected_components.cpp [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+#include <cugraph/graph_functions.hpp>
+
+#include <optional>
+
+namespace {
+
+struct scc_functor : public cugraph::c_api::abstract_functor {
+  raft::handle_t const& handle_;
+  cugraph::c_api::cugraph_graph_t* graph_;
+  bool do_expensive_check_;
```
Comment (seunghwak) [inline cpp/src/c_api/strongly_connected_components.cpp:37]: Add "{}"

File: cpp/tests/c_api/mg_eigenvector_centrality_test.c [hunk 4]
```diff
@@ -106,16 +107,16 @@ int test_eigenvector_centrality(const cugraph_resource_handle_t* handle)
...
+                                             h_src,
+                                             h_dst,
+                                             h_wgt,
+                                             h_result,
+                                             num_vertices,
+                                             num_edges,
+                                             TRUE,
+                                             alpha,
+                                             epsilon,
+                                             max_iterations);
```
Comment (seunghwak) [inline cpp/tests/c_api/mg_eigenvector_centrality_test.c:119]: Isn't the updates in this file from the eigenvector C API PR?

Comment (ChuckHastings) [reply cpp/tests/c_api/mg_eigenvector_centrality_test.c:119]: I had to modify the call to `create_mg_test_graph` and `create_test_graph` (the SG version) to add the is_symmetric parameter.  This required me to change all of the C unit tests.  Not quite sure why the indentation changed.  Perhaps an upstream change in the clang format parameters?

File: cpp/src/c_api/weakly_connected_components.cpp (hunk 2 above)
Comment (ChuckHastings) [reply cpp/src/c_api/weakly_connected_components.cpp:37]: In latest push

File: cpp/src/c_api/strongly_connected_components.cpp (hunk 3 above)
Comment (ChuckHastings) [reply cpp/src/c_api/strongly_connected_components.cpp:37]: in latest push

File: cpp/src/c_api/strongly_connected_components.cpp (hunk 1 above)
Comment (ChuckHastings) [reply cpp/src/c_api/strongly_connected_components.cpp]: Fixed in latest push

**Comment by BradReesWork** [comment]
@gpucibot merge

**Pull request merged**